src/transaction/log_replication.cpp

```diff
@@ -272,15 +272,39 @@ namespace cublog
 	    m_most_recent_trantable_snapshot_lsa.store (m_redo_lsa);
 	    break;
 	  case LOG_MVCC_UNDO_DATA:
-	    read_and_bookkeep_mvcc_vacuum<LOG_REC_MVCC_UNDO> (header.type, header.back_lsa, m_redo_lsa, true);
+	  {
+	    m_redo_context.m_reader.advance_when_does_not_fit (sizeof (LOG_REC_MVCC_UNDO));
```

Comment:
Is the code for handling alignment and record not fitting particular to this change ?
If not, I suggest have separate PR as a fix.
What about other records ?

Later edit : I noticed this is moved code and it is not a consitent rule for handling it.
It looks like a general problem which could be handled before switch-case.
It looks we already do "reinterpret_copy_and_add_align", we need to add  "advance_when_does_not_fit" before it, and get rid of this code for each record in switch-case.

Reply:
It cannot be implemented before switch because it depends on the type of log record. But you are right that the usage of both alignment and advancement functions is not consistent. I will move them to a separate patch.